### ISSUE TITLE
1.15.x: hubble-ui: envoy refuses to connect to nginx

### ISSUE THREAD

**Issue by cornfeedhobo** [cilium/cilium#29985]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When enabling the hubble ui and associated ingress, [or gateway](https://github.com/cilium/cilium/issues/29563), the UI errors with the following request and response:

```
POST /api/ui.UI/GetControlStream 400
```
```plain
grpc-message: gRPC requires HTTP/2
```

I am able to port forward the ui pod locally and that works correctly, which seems to also verify this is happening at the envoy layer.

I made some rudimentary progress, but it's not fixed completely. Nginx's `proxy_pass` directive only supports up to http 1.1, so I'm guessing that the nested grpc path `/api/ui.UI` is not satisfying envoy's required 2.x. By adding the following directive to nginx's config, I was able to get the hubble UI to load again

```plain
        location /api/ui.UI {
            grpc_hide_header Access-Control-Allow-Origin;
            grpc_pass grpc://127.0.0.1:8090;
        }
```

This is not a complete fix. I need someone with domain specific knowledge to take a look please.

### Cilium Version

1.15.0-rc.0

### Kernel Version

6.6.6-1-default

### Kubernetes Version

v1.26.10+k3s2

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by squeed** [comment]
@cornfeedhobo thanks for the report. (I'm just the triager, sadly I won't be able to fix this). But, a few initial questions:

1. Did you create the Gateway resources, or are they coming from a helm chart? (pardon my ignorance)
2. Is there another issue specifically for the bad nginx config? I didn't see one, but it seems like there are a few related issues.

Thanks

**Comment by cornfeedhobo** [comment]
@squeed 

1) I created the gateway resources myself. I don't think the helm chart had support for that, at the time that I was testing. But I also tested with gateway support disabled and only using the ingress controller and the helm-generated ingress resources, which also suffered from the same problem.

2) I only know about this one and the one I linked to, #29563.

P.S. I'm too busy to reproduce this again right now, but someone on the cilium team should be able to reproduce this quickly. I know at least two other users have reported the issue.

Thanks!
